addon/mixins/handles-http-status-error.js

```diff
@@ -0,0 +1,57 @@
+import Ember from 'ember'
+
+export default Ember.Mixin.create({
+  i18n: Ember.inject.service(),
+  flashMessages: Ember.inject.service(),
+
+  /**
+  * Handles various types of server errors denoted by
+  * HTTP status codes.
+  *
+  * It follows the convention of Ember error event handlers,
+  * returning true IF the client of this method must handle
+  * the error, and false if the error has been handled.
+  *
+  * @return {boolean}  True when you need to handle the error yourself.
+  *                    False when error is taken care of by this mixin.
+  */
+  handleHttpStatusError(error) {
+    return !(
+      this._handleUnauthorizedError(error) ||
+      this._handleConflictError(error) ||
+      this._handleServiceUnavailableError(error)
+    )
+  },
+
+  _handleUnauthorizedError(error) {
+    if (this._flashOnError(error, '401', 'flash.http_codes.unauthorized')) {
```

Comment:
En ting jeg kom til å tenke på. Kan det være problematisk at vi refererer til i18n strenger her, uten at echo-ember-common har i18n? Ville det være verdt å vurdere å la den i18n som er i felles koden også være felles, eller vil vi heller ha muligheten til å endre oversettelsene uavhengig av hverandre?

Reply:
Liker tanken med at 18n oversettelser kunne vært definert i common. Bare ikke helt sikker på hva som er enkleste måte å gjøre det på. Nærmeste jeg kom var [dette](https://github.com/jamesarosen/ember-i18n/issues/256), men det hjelper ikke helt. Ser for meg at det verste som skjer hvis man legger til en ny feilhåndtering her uten å oppdatere språkfil i app-ene er at man får ut en oversettelse-ikke-funnet melding tenker jeg.